kibana/lib/openshift_logging_plugin/public/hack.js

```diff
@@ -0,0 +1,67 @@
+import chrome from 'ui/chrome';
+import { uiModules } from 'ui/modules';
+const app = uiModules.get('apps/openshift_logging_plugin/index');
+
+function redirectOnSessionTimeout($window) {
+    document.cookie = "name=_oauth_proxy; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/;"
+    const APP_ROOT = `${chrome.getBasePath()}`;
+    const path = chrome.removeBasePath($window.location.pathname);
+    window.location.href = APP_ROOT.substring(0,path.length + 1) + '/oauth/sign_out'
+}
+
+app.factory('errorInterceptor', function ($q, $window) {
+
+    return {
+        responseError: function (response) {
+          console.log(response)
+          if (response.status == 401) { 
+              console.log("Session expired. Redirecting to sign_off to refresh")
+              redirectOnSessionTimeout($window);
+          }
+
+          // If unhandled, we just pass the error on to the next handler.
+          return $q.reject(response);
+        }
+    };
+});
+
+/**
+ * Make sure that we add the interceptor to the existing ones.
+ */
+app.config(function($httpProvider) {
+    $httpProvider.interceptors.push('errorInterceptor');
+});
+
+/**
+ * Setup a wrapper around fetch so that we can
+ * handle session timeouts on ajax calls made
+ * by the kfetch component
+ * @param $window
+ */
+function setupResponseErrorHandler($window) {
+    if (!window.fetch) {
+        return;
+    }
+
+    const nativeFetch = window.fetch;
+    window.fetch = (url, config) => {
+        return nativeFetch(url, config)
+            .then(async(result) => {
```

Comment:
Do JS Promises not follow the `reject` path if they fail?

Reply:
:man_shrugging: copy / paste here

Reply:
Except from https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API: